marklogic-data-hub/src/test/java/com/marklogic/hub/flow/RunFlowWithSameJobIdTest.java

```diff
@@ -2,40 +2,42 @@
         installProjectInFolder("test-projects/simple-customer-flow");
-        FlowRunnerImpl flowRunner = new FlowRunnerImpl(getHubConfig(), new FlowManagerImpl(getHubConfig()));
 
-        flowRunner.runFlow(new FlowInputs("simpleCustomerFlow", "1").withJobId("abc123"));
-        flowRunner.awaitCompletion();
+        runFlow(new FlowInputs("simpleCustomerFlow", "1").withJobId("abc123"));
 
         ObjectNode firstFlowResponse = (ObjectNode) jobDocManager.getLatestJobDocumentForFlow("simpleCustomerFlow");
-        String timeStarted1= firstFlowResponse.get("job").get("timeStarted").asText();
-        String timeEnded1= firstFlowResponse.get("job").get("timeEnded").asText();
-        String stepsStartTime1= firstFlowResponse.get("job").get("stepResponses").get("1").get("stepStartTime").asText();
-        String stepsEndTime1= firstFlowResponse.get("job").get("stepResponses").get("1").get("stepEndTime").asText();
+        LocalDateTime timeStarted1 = parseDateTime(firstFlowResponse.get("job").get("timeStarted").asText());
+        LocalDateTime timeEnded1 = parseDateTime(firstFlowResponse.get("job").get("timeEnded").asText());
+        LocalDateTime stepStartTime1 = parseDateTime(firstFlowResponse.get("job").get("stepResponses").get("1").get("stepStartTime").asText());
+        LocalDateTime stepEndTime1 = parseDateTime(firstFlowResponse.get("job").get("stepResponses").get("1").get("stepEndTime").asText());
 
-        flowRunner.runFlow(new FlowInputs("simpleCustomerFlow", "1").withJobId("abc123"));
-        flowRunner.awaitCompletion();
+        runFlow(new FlowInputs("simpleCustomerFlow", "1").withJobId("abc123"));
 
         ObjectNode secondFlowResponse = (ObjectNode) jobDocManager.getLatestJobDocumentForFlow("simpleCustomerFlow");
-        String timeStarted2= secondFlowResponse.get("job").get("timeStarted").asText();
-        String timeEnded2= secondFlowResponse.get("job").get("timeEnded").asText();
-        String stepsStartTime2= secondFlowResponse.get("job").get("stepResponses").get("1").get("stepStartTime").asText();
-        String stepsEndTime2= secondFlowResponse.get("job").get("stepResponses").get("1").get("stepEndTime").asText();
-
-        Boolean result = timeStarted1.equals(timeStarted2) || timeEnded1.equals(timeEnded2) || stepsStartTime1.equals(stepsStartTime2) || stepsEndTime1.equals(stepsEndTime2);
-        assertEquals(false,result);
+        LocalDateTime timeStarted2 = parseDateTime(secondFlowResponse.get("job").get("timeStarted").asText());
+        LocalDateTime timeEnded2 = parseDateTime(secondFlowResponse.get("job").get("timeEnded").asText());
+        LocalDateTime stepStartTime2 = parseDateTime(secondFlowResponse.get("job").get("stepResponses").get("1").get("stepStartTime").asText());
+        LocalDateTime stepEndTime2 = parseDateTime(secondFlowResponse.get("job").get("stepResponses").get("1").get("stepEndTime").asText());
+
+        assertTrue(timeEnded1.isAfter(timeStarted1));
```

Comment:
@AkshayBajajML One note about the assertions here - we generally want each assertion to verify a single fact. The problem with an assertion that verifies several things is that if one fails, it's not apparent which one it is, which makes debugging a little more difficult.